src/app/modules/main/chat_section/model.nim

```diff
@@ -227,6 +263,18 @@ QtObject:
         defer: index.delete
         self.dataChanged(index, index, @[ModelRole.CategoryOpened.int])
 
+  proc changeCategoryItemIfHiddenBecauseNoPermitted*(self: Model, id: string) =
+    if id == "":
+      return
+    let index = self.getItemIdxById(id)
+    if index == -1:
+      return
+    if not self.items[index].isCategory:
+      return
+    let modelIndex = self.createIndex(index, 0, nil)
+    defer: modelIndex.delete
+    self.dataChanged(modelIndex, modelIndex, @[ModelRole.ShouldBeHiddenBecausePermissionsAreNotMet.int])
```

Comment:
You didn't change anything actually. I think you forgot to change the value.

Also, we only want to hide the category if all the channels in it are hidden. So we need a loop to check the channels in the category here.

Reply:
I renamed this function and also added a few comments for better understanding.
Every item has ShouldBeHiddenBecausePermissionsAreNotMet role. It is a special, helper role that I added which is a combination of other roles. This role is used to filter in QML. The value of this role is computed by itemShouldBeHiddenBecauseNotPermitted() function for both normal items and section items.
Whenever data with roles: HideIfPermissionsNotMet, ViewOnlyPermissionsSatisfied, ViewAndPostPermissionsSatisfied, are changed, also we trigger refreshing data with ShouldBeHiddenBecausePermissionsAreNotMet role.
This works correctly as I showed on the demo.